src/hooks/useSanctionedAddress.ts

```diff
@@ -0,0 +1,22 @@
+import { useGetIsSanctionedQuery } from '@/store/ofac'
+import useSafeAddress from './useSafeAddress'
+import useWallet from './wallets/useWallet'
+import { skipToken } from '@reduxjs/toolkit/query/react'
+
+export const useSanctionedAddress = () => {
+  const wallet = useWallet()
+  const safeAddress = useSafeAddress()
+
+  const { data: isWalletSanctioned } = useGetIsSanctionedQuery(wallet ? wallet.address : skipToken)
+
+  const { data: isSafeSanctioned } = useGetIsSanctionedQuery(safeAddress !== '' ? safeAddress : skipToken)
+
+  if (isSafeSanctioned) {
+    return safeAddress
+  }
+  if (isWalletSanctioned) {
+    return wallet?.address
+  }
```

Comment:
`data` contains the error message in case the `useGetIsSanctionedQuery` throws so I think it would falsely say that an address is sanctioned if the network request fails.

Reply:
I think that is not true. The error seems to be transformed and returned in the `error` field of `useGetIsSanctionedQuery`

Reply:
Should we remove the data assignment on error then inside the `ofacApi` query?

`return { error: { status: 'CUSTOM_ERROR', data: (error as Error).message } }`